### ISSUE TITLE
rsyslog relp fails to load on config that works in earlier rsyslog versions

### ISSUE THREAD

**Issue by Rumbles** [rsyslog/rsyslog#3106]
I am running a few servers using rsyslog, they're running ubuntu 14 or 16, and have the standard rsyslog from the packages. My config is in the old format, and I am looking at upgrading it, but I need the functionality from later versions of rsyslog than is available in the standard repos, so I'm looking at using the adiscon repo instead. I installed the repo and restarted rsyslog and it failed to start imrelp with the following messages in syslog:

```
rsyslogd: imrelp: no RELP listener defined, module can not run. [v8.38.0 try http://www.rsyslog.com/e/2172 ]
rsyslogd: activation of module imrelp failed [v8.38.0 try http://www.rsyslog.com/e/-3 ]
```

I managed to get it working with the new style config, but I would have expected the the old style config should work in all versions of rsyslog 8

I guess something has changed and I have missed it, but I am at a loss at the moment. I'm concerned it may be a bug in the way the old config is handled.

### Expected behaviour

I am loading the input with the following config:

```
$ModLoad imrelp # Load the input module ('im') 'relp'
$InputRELPServerRun 25014 # Set the port to 25014
```

It fails to load with the above message, switching back to 8.16 from standard repos works fine

### Actual behavior

Should load imrelp and listen on the provided port

### Environment
- rsyslog version:
```
ii  rsyslog                          8.38.0-0adiscon1xenial1                    amd64        a rocket-fast system for log processing                         
ii  rsyslog-relp                     8.38.0-0adiscon1xenial1                    amd64        RELP protocol support for rsyslog
```

- platform: Ubuntu 16.04
- for configuration questions/issues, include rsyslog.conf and included config files (Trimmed most of the comments, otherwise this is all my config on the central server)

```$MaxMessageSize 64k

#################
#### MODULES ####
#################
$ModLoad imuxsock # provides support for local system logging
$ModLoad imklog   # provides kernel logging support
#$ModLoad immark  # provides --MARK-- message capability

# Input file
$Modload imfile

#Include the FQDN on logs
$PreserveFQDN on

# provides UDP syslog reception
#$ModLoad imudp
#$UDPServerRun 514

# provides TCP syslog reception
#$ModLoad imtcp
#$InputTCPServerRun 514

# Enable non-kernel facility klog messages
$KLogPermitNonKernelFacility on

###########################
#### GLOBAL DIRECTIVES ####
###########################

#
# Use traditional timestamp format.
# To enable high precision timestamps, comment out the following line.
#
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat

# Filter duplicated messages
$RepeatedMsgReduction off

#
# Set the default permissions for all log files.
#
$FileOwner syslog
$FileGroup syslog
$FileCreateMode 0664
$DirCreateMode 0775
$Umask 0002
$PrivDropToUser syslog
$PrivDropToGroup syslog

#
# Include all config files in /etc/rsyslog.d/
#
$IncludeConfig /etc/rsyslog.d/*.conf

$ModLoad imrelp # Load the input module ('im') 'relp'
$InputRELPServerRun 25014 # Set the port to 25014

$template CustomForwardFormat,"<%PRI%>%TIMESTAMP:::date-pgsql% - %msg:::sp-if-no-1st-sp%%msg%\n"

# Handle the internal syslog event messages from remote hosts
$template DynaFile,"/srv/rsyslog/%$YEAR%/%HOSTNAME%/%$MONTH%/%$DAY%/%syslogfacility-text%.log"
*.*;local7,local6.none  ?DynaFile;CustomForwardFormat

# Handle the local facility log messages from remote hosts
$template DynaFileLocal,"/srv/rsyslog/%$YEAR%/%HOSTNAME%/%$MONTH%/%$DAY%/%syslogtag:1:32%.log"
local7,local6.* ?DynaFileLocal;CustomForwardFormat
# Drop all messages that are not from the local machine, so they don't go in to local syslog
if ($hostname != 'rsyslog.myhostname.net') then stop

#Less relevant config

$InputFileName /var/log/salt/minion
$InputFileTag minion
$InputFileStateFile minion-state

$InputFileSeverity  info
$InputFileFacility  local6
$InputRunFileMonitor


$InputFilePollInterval 1

auth,authpriv.*     /var/log/auth.log
*.*;auth,authpriv.none;local6.none;local7.none;local5.none    -/var/log/syslog
#cron.*       /var/log/cron.log
#daemon.*     -/var/log/daemon.log
kern.*        -/var/log/kern.log
#lpr.*        -/var/log/lpr.log
mail.*        -/var/log/mail.log
#user.*       -/var/log/user.log

#
# Logging for the mail system.  Split it up so that
# it is easy to write scripts to parse these files.
#
#mail.info      -/var/log/mail.info
#mail.warn      -/var/log/mail.warn
mail.err      /var/log/mail.err

#
# Logging for INN news system.
#
news.crit     /var/log/news/news.crit
news.err      /var/log/news/news.err
news.notice     -/var/log/news/news.notice

#
# Some "catch-all" log files.
#
#*.=debug;\
# auth,authpriv.none;\
# news.none;mail.none -/var/log/debug
#*.=info;*.=notice;*.=warn;\
# auth,authpriv.none;\
# cron,daemon.none;\
# mail,news.none    -/var/log/messages

#
# Emergencies are sent to everybody logged in.
#
*.emerg                                :omusrmsg:*
```

**Comment by alorbach** [comment]
Can you run "ls -al /usr/lib/rsyslog/" to see if all modules were updated properly? They should be from the same build.

**Comment by alorbach** [comment]
And I would recommend to use new configuration style for relp module loading and input: 

module(load="imrelp") 

input(type="imrelp" port="20514") {
...
}

**Comment by alorbach** [comment]
Thanks for pointing that out. 
We will put this on the to look into list for later.

**Comment by Rumbles** [comment]
Thanks @alorbach I see a few emails but only the one message on here, does that mean you were able to reproduce?
I reverted my machine back to 8.16 for the time being as it wasn't behaving, I can test if you need me to.

**Comment by alorbach** [comment]
it means my earlier responses were wrong ;)

**Comment by davidelang** [comment]
On Mon, 8 Oct 2018, Andre Lorbach wrote:

> And I would recommend to use new configuration style for relp module loading and input:
>
> module(load="imrelp")
>
> input(type="imrelp" port="20514") {
> ...
> }

while this is recommended, if the old syntax doesn't work, it's a bug we need to 
fix.

David Lang

**Comment by alorbach** [comment]
You are right David, found this one. Was caused by https://github.com/rsyslog/rsyslog/commit/32b71daa8aadb8f16fe0ca2945e54d593f47a824 

fix will be merged once testbench is complete..
